### PR TITLE
correct py-bx-python@0.8.8 python dependency

### PR DESCRIPTION
**Pull request by mikerenfro**
py-bx-python@0.8 in the develop branch fails to build with Python 3.10. The [py-bx-python release notes](https://github.com/bxlab/bx-python/releases) show support for 3.8 and earlier. From initial testing, this built on 3.9 as well, but I'm sticking with what's documented upstream.

### CODE REVIEW

**Review by adamjstewart — DISMISSED** [review]
For projects like this that are tightly coupled to the CPython API I like to use the wheels available on PyPI as guides for what versions are supported. Up to you whether you also want to limit 0.9.0 in this way.

**Comment by mikerenfro** [comment]
> For projects like this that are tightly coupled to the CPython API I like to use the wheels available on PyPI as guides for what versions are supported. Up to you whether you also want to limit 0.9.0 in this way.

I don't have a real strong opinion on it, but I've put explicit version ranges on both 0.8.8 and 0.9.0 now.

File: var/spack/repos/builtin/packages/py-bx-python/package.py [hunk 1]
```diff
@@ -15,11 +15,15 @@ class PyBxPython(PythonPackage):
...
+    # work with which releases.
+    depends_on("python@3.7:3.11", when="@=0.9.0", type=("build", "run"))
+    depends_on("python@3.5:3.8", when="@=0.8.8", type=("build", "run"))
     depends_on("py-setuptools", type="build")
     depends_on("py-cython", type="build")
     depends_on("py-numpy", type=("build", "run"))
     depends_on("py-six", when="@:0.8.9", type=("build", "run"))
     # py-python-lzo is listed as a dependency in `tox.ini` rather than in `setup.cfg`
     depends_on("py-python-lzo@1.14:", type=("build", "run"))
+    depends_on("zlib", type=("link", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bx-python/package.py]: "run" doesn't make sense here, there's nothing to add to `PATH`

Comment (mikerenfro) [reply var/spack/repos/builtin/packages/py-bx-python/package.py]: Fixed.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**